src/main/java/backend/Logic.java

```diff
@@ -232,7 +234,12 @@ public void removeUnusedModels(Set<String> reposInUse) {
     }
 
     public ImmutablePair<Integer, Long> updateRemainingRate(ImmutablePair<Integer, Long> rateLimits) {
-        uiManager.updateRateLimits(rateLimits);
+        uiManager.updateRemainingRate(rateLimits);
+        return rateLimits;
+    }
+
+    public ImmutablePair<Integer, Long> updateSyncRefreshRate(ImmutablePair<Integer, Long> rateLimits) {
```

Comment:
It should still be helpful to have a header comment for this method explaining its effect.